ls1intum/Artemis
# [Exam] Disable hints for exam

### Checklist

### Motivation and Context
Code editor issues multiple requests to fetch hints. Also hints are currently not used so they can be deactivated for exam exercises.

### Description

- [x] Disable creation of hints for exam exercises on server side
- [x] Disable attaching hints to exam exercises in the PUT request
- [x] Hide the "Manage Hints" button in the programming exercise details page for exam exercises
- [x] Refactor code editor to issue only one request for hints
- [x] Disable fetching of hints for exam mode code editor

### Steps for Testing


_Button disabled_
1. Create an exam programming exercise
2. Navigate to the details page of this exercise
--> Manage hints button should not be displayed

_Hints load once for normal programming exercises_
1. Create a normal programming exercise with code editor enabled
2. Create hints for it
3. Open the online code editor and open the network dev tool
--> Hints should be loaded correctly but only once

_Hints are not fetched for exam exercises_
1. Create a exam programming exercise with code editor enabled
2. Participate in the exam and open the network dev tool
--> No hint request should be issued
